.github/workflows/lint.yaml

```diff
@@ -0,0 +1,57 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v4
+        with:
+          go-version-file: go.mod
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.64.6
+          working-directory: ${{ inputs.cwd }}
+
+      - name: staticcheck
+        uses: dominikh/staticcheck-action@v1
+        with:
+          version: 2025.1.1
+          install-go: false
+          working-directory: ${{ inputs.cwd }}
+          cache-key: ${{ inputs.cwd }}
+
+      - name: Run allowtags
+        working-directory: ${{ inputs.cwd }}
+        run: |
+          go install github.com/ashmrtn/allowtags@v0.0.2
+          allowtags --allow-key json       \
+            ./...
+
+      - name: Run tint
+        run: |
+          go install github.com/meain/tint@latest
+          tint lint ${{ inputs.cwd }}
+
+      # Upload the original go test output as an artifact for later review.
+      - name: Upload test log
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: unit-test-log
+          path: testlog/*
+          if-no-files-found: error
+          retention-days: 14
```

Comment:
nit: this step seems unnecessary (and may even fail consistently) as there's no logs generated during this job

```suggestion
```